src/WebJobs.Script/Description/Node/Script/functionTemplate.js

```diff
@@ -5,6 +5,9 @@ var f = require('{0}'),
     util = require('util');
 
 return function (context, callback) {{
+    // TEMP HACK: workaround for https://github.com/tjanczuk/edge/issues/325
+    setImmediate(() => {{}});
```

Comment:
My vote would be that if all our testing shows that this new workaround addresses all the previous issues in addition to fixing promises, then we just go with it and remove the old workaround. @brettsam Does this look like it addresses all the scenarios you tested before?